src/SendGrid/SendGridClient.cs

```diff
@@ -129,168 +144,25 @@ public enum Method
-        /// <param name="requestHeaders">A dictionary of request headers</param>
-        public void AddRequestHeader(Dictionary<string, string> requestHeaders)
-        {
-            this.RequestHeaders = (this.RequestHeaders != null)
-                    ? this.RequestHeaders.Union(requestHeaders).ToDictionary(pair => pair.Key, pair => pair.Value) : requestHeaders;
-        }
-
-        /// <summary>
-        /// Set or update the UrlPath
-        /// </summary>
-        /// <param name="urlPath">The endpoint without a leading or trailing slash</param>
-        public void SetUrlPath(string urlPath)
-        {
-            this.UrlPath = urlPath;
-        }
-
-        /// <summary>
-        /// Get the urlPath to the API endpoint
-        /// </summary>
-        /// <returns>Url path to the API endpoint</returns>
-        public string GetUrlPath()
-        {
-            return this.UrlPath;
-        }
-
-        /// <summary>
         /// Add the authorization header, override to customize
         /// </summary>
         /// <param name="header">Authorization header</param>
         /// <returns>Authorization value to add to the header</returns>
         public virtual AuthenticationHeaderValue AddAuthorization(KeyValuePair<string, string> header)
         {
-            string[] split = header.Value.Split(new char[0]);
+            string[] split = header.Value.Split();
             return new AuthenticationHeaderValue(split[0], split[1]);
         }
 
         /// <summary>
-        /// Add the version of the API, override to customize
-        /// </summary>
-        /// <param name="version">Version string to add to the URL</param>
-        public void SetVersion(string version)
-        {
-            this.Version = version;
-        }
-
-        /// <summary>
-        /// Get the version of the API, override to customize
-        /// </summary>
-        /// <returns>Version of the API</returns>
-        public string GetVersion()
-        {
-            return this.Version;
-        }
-
-        /// <summary>
         /// Make the call to the API server, override for testing or customization
         /// </summary>
-        /// <param name="client">Client object ready for communication with API</param>
         /// <param name="request">The parameters for the API call</param>
         /// <param name="cancellationToken">Cancel the asynchronous call</param>
         /// <returns>Response object</returns>
-        public async Task<Response> MakeRequest(HttpClient client, HttpRequestMessage request, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<Response> MakeRequest(HttpRequestMessage request, CancellationToken cancellationToken = default(CancellationToken))
         {
-            HttpResponseMessage response = await client.SendAsync(request, cancellationToken).ConfigureAwait(false);
+            HttpResponseMessage response = await client.SendAsync(request, cancellationToken);
```

Comment:
Why drop the `.ConfigureAwait(false);`? I believe we had async issues before without this call. Please see this thread: https://github.com/sendgrid/sendgrid-csharp/issues/362